src/background.ts

```diff
@@ -6,15 +6,34 @@
     const calendar_name = "AutoCalendar Demo";
     const autocalendar = new AutoCalendar();
 
-    chrome.runtime.onMessage.addListener(
-        async (request, sender, sendResponse) => {
-            if (request.action === "sync_canvas") {
+    let prev_execution = null;
+    let isRunning = false;
```

Comment:
This flag is almost certainly unnecessary; you should be able to refactor it out. Flags are only necessary within complex algorithms usually, they clutter up the code a lot so should be avoided when possible.

Reply:
Currently there is no better way to ensure one sync is happening at a time, and the flag is required by the front-end to behave properly depending on if background is running or not.